### ISSUE TITLE
Unpacked Backup is Empty

### ISSUE THREAD

**Issue by zedseven** [nelenkov/android-backup-extractor#76]
Extracting an encrypted backup yields a *.tar file that is exactly 1024 bytes long, all of which are 00. I have tried backing up a few apps, all of which to the same result.

These are the exact commands I have run:
```
./adb.exe backup -f ./com.fedex.ida.android.ab -noapk com.fedex.ida.android

java -jar ./abe-all.jar unpack "com.fedex.ida.android.ab" "com.fedex.ida.android.tar" 1111
```

And running `java -version` yields:
```
java version "1.8.0_221"
Java(TM) SE Runtime Environment (build 1.8.0_221-b11)
Java HotSpot(TM) Client VM (build 25.221-b11, mixed mode)
```

[I have attached a sample app backed up from my old phone (LG G6) encrypted with the password '1111'](https://github.com/nelenkov/android-backup-extractor/files/3853793/com.fedex.ida.android.Failed.Backup.Extraction.zip)

**Comment by marrakchino** [comment]
Hi,
I just ran into the same issue.

I have a slightly different configuration : 
```
[23:28] $ java -version
openjdk version "11.0.5" 2019-10-15
OpenJDK Runtime Environment (build 11.0.5+10-post-Ubuntu-2ubuntu116.04)
OpenJDK 64-Bit Server VM (build 11.0.5+10-post-Ubuntu-2ubuntu116.04, mixed mode, sharing)
```

The commands I performed : 
```
[23:23] ~/tmp$ java -jar ~/Downloads/abe.jar unpack packpack.ab packpack.tar 1234
Calculated MK checksum (use UTF-8: true): <hash>
3% 
1024 bytes written to packpack.tar.
[23:32] ~/tmp$ java -jar ~/Downloads/abe.jar unpack packpack.ab packpack.tar
[23:32] ~/tmp$ ls -lh packpack.tar 
-rw-rw-r-- 1 marrakchino marrakchino 1.0K Mar 15 23:28 packpack.tar
```

The same commands yield a 1.0K tar _empty_ archive (nothing extractable).
I used a Samsung Galaxy S8 (Android 9, API 28) phone in the process.

Any updates ?
Best

**Comment by stachenov** [comment]
Same here. I was hoping to export MS Outlook configuration this way, so I could then import it on a PC. Not sure if that is possible at all, but couldn't even get past the export step: got tar of 1024 zero bytes.

**Comment by stachenov** [comment]
Looking at the source file, it's hard to believe it contains any useful data. It's just too small for that (549 bytes). So I wouldn't blame the extractor for it. It's that `adb` just can't make backups of certain apps on certain phones. Why, I have no idea.

**Comment by nelenkov** [comment]
Apps can disable backup of their data. Different Android versions handle this differently, some produce empty backups (0 bytes), some output header with no data. ADB backup is being deprecated, so no guarantees. 
You could try enabling developer options and setting a backup password there, but very much depends on device, Android version, and app settings combintation.

**Comment by stachenov** [comment]
Right, that's exactly what I'm thinking. I tried to make a full backup of all apps and found that data for some apps is present, while for others it's not.

And it is certainly *not* Extractor's fault because it is obvious from 549-byte size of the original `*.ab` file that it was empty to begin with. No way Extractor can extract some useful data from an empty file. Same with the file attached by the OP (it's also 549 bytes).

**Comment by gsgrippal** [comment]
com.fedex.ida.android does not allow backup, you have to modify apk to work

**Comment by superbonaci** [comment]
@zedseven close this issue it's the app's fault.
